lib/gems/pending/util/require_with_logging.rb

```diff
@@ -30,16 +31,28 @@ def log_require(path, mode, timing = nil)
     $req_depth += 1 if mode == :enter || mode == :reenter
   end
 
+  def log_require_time
+    Time.now
+  end
+
+  def log_require_allocations
+    GC.stat[:total_allocated_objects]
+  end
+
+  def log_require_mb
+    Sys::ProcTable.ps(Process.pid).smaps.pss / 1_000_000.0
```

Comment:
yeah, sys proctable doesn't guarantee each platform will have the same attributes/methods on the object returned from `ps`, so the OSX and other non-linux OS's will not have smaps.

Maybe, we should raise on non-Linux platforms.